homestar-runtime/src/settings.rs

```diff
@@ -22,6 +24,55 @@ pub struct Monitoring {
     process_collector_interval: u64,
 }
 
+#[derive(Default, Clone, Debug, Deserialize)]
+/// Configure how the Network keypair is generated or using an existing one
+pub(crate) enum PubkeyConfig {
+    #[default]
+    #[serde(rename = "random")]
+    Random,
+    /// Seed string is hashed with SHA-256 to produce the ed25519 secret key.
+    #[serde(rename = "seed")]
+    GenerateFromSeed(String),
+    /// File path to a PEM encoded ed25519 key
+    #[serde(rename = "path")]
+    Existing(String),
+}
+
+
+impl PubkeyConfig {
+    /// Produce a Keypair using the given configuration. Consumes `self` to avoid keeping secrets laying around.
+    fn generate_keypair(self) -> Result<identity::Keypair, DecodingError> {
```

Comment:
instead of decoding error, let's use `anyhow::Result` as per the runtime. I want to start labeling errors and bubbling up to anyhow, but we can tackle that in an error-specific PR. If you use anyhow here, you can just use `?` or `anyhow::context` to provide the decodingerror as a contextual error chain.